src/charm.py

```diff
@@ -95,6 +120,58 @@ def _on_peer_relation_changed(self, event):
         """
         self._update(event)
 
+    @log_event_handler(logger)
+    def _on_update_status(self, event):
+        """Handle `update-status` events.
+
+        Args:
+            event: The `update-status` event triggered at intervals.
+        """
+        try:
+            self._validate()
+        except ValueError:
+            return
+
+        all_valid_plans = True
+        for container_name in list(CONTAINERS.keys()):
+            if not CONTAINERS[container_name]:
+                continue
+
+            container = self.unit.get_container(container_name)
+            valid_pebble_plan = self._validate_pebble_plan(container, container_name)
+            logger.info(f"validating pebble plan for {container_name}")
+            if not valid_pebble_plan:
+                all_valid_plans = False
```

Comment:
Would it be worth `logger.debug(f"failed to validate pebble plan for {container_name}, attempting creation again")` or something like that so you can work out how often this pebble plan failure is occuring?